Sources/Core/AWSClientRuntime/EventStream/AWSMessageDecoder.swift

```diff
@@ -88,6 +103,38 @@ extension AWSEventStream {
             return message
         }
 
+        // Responsible for waiting on the initial response.
+        // It uses Swift's concurrency model to asynchronously return the data.
+        public func awaitInitialResponse() async -> Data? {
+            // The 'withCheckedContinuation' function is used to bridge asynchronous code
+            // that doesn't use Swift's concurrency model with code that does.
+            return await withCheckedContinuation { continuation in
+                // Here, we attempt to retrieve the initial response.
+                // Once the data is retrieved (or determined to be nil),
+                // the continuation is resumed with the result.
+                retrieveInitialResponse { data in
+                    continuation.resume(returning: data)
```

Comment:
It is required, in all cases, that the continuation gets called (it's a rather large memory leak and a frozen path of execution if it doesn't.)

Considering that `retrieveInitialResponse()` will in some cases save the continuation to an instance var, how do we guarantee that it gets called in every case?

Reply:
Agreed in separate conversation that immediately after running the continuation stored in `didProcessInitialMessage`, set it to nil so it is released and it cannot be called again.